fairlearn/_input_validation.py

```diff
@@ -82,9 +85,10 @@ def _validate_and_reformat_input(X, y=None, expect_y=True, enforce_binary_sensit
         sensitive_features = \
             _compress_multiple_sensitive_features_into_single_column(sensitive_features)
 
-    if enforce_binary_sensitive_feature:
-        if len(np.unique(sensitive_features)) > 2:
-            raise ValueError(_SENSITIVE_FEATURES_NON_BINARY_ERROR_MESSAGE)
+    if group_warn_threshold is not None:
```

Comment:
I think it should default to 2, so we keep the previous behaviour as a default.

Reply:
I don't think we should check for the number of sensitive feature values; instead, we should add a warning if `true_dim>4` on:
https://github.com/fairlearn/fairlearn/blob/78d18451afadc1365af47e8b5a502c28c2a51db4/fairlearn/reductions/_grid_search/_grid_generator.py#L22
(we could also consider a more sophisticated condition that would be based on both the grid size and the grid dimension, but I'm not exactly sure what it should be).

Reply:
OK, I adjusted it according to your specification @MiroDudik . Resolving this since it's based on an old version of the code.